### PR TITLE
Add CeloscanProvider

### PR DESCRIPTION
**Pull request by anarkrypto**
This PR implements CeloscanProvider, a wrapper for EtherscanProvider. Thus, Celo users can access functionalities such as getHistory (obtaining transactions from a given account).

This provider uses the Celo explorer API for the 3 network options:

Celo Mainnet: https://explorer.celo.org/mainnet
Alfajores:        https://explorer.celo.org/alfajores
Baklava:         https://explorer.celo.org/baklava

- [x] Support to apiKey
- [x] Unit test
- [x] README example

### CODE REVIEW

**Comment by anarkrypto** [comment]
Unit test example:

```bash
ACCOUNT=0x4443381f564ea8216b25ddd25710fe301f100634 NETWORK=alfajores yarn test:history
```

File: src/lib/CeloscanProvider.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    getBaseUrl(): string {
+        switch (this.network ? this.network.name : "invalid") {
+            case "celo":
+                return "https:/\/explorer.celo.org/mainnet";
+            case "alfajores":
+                return "https:/\/explorer.celo.org/alfajores";
+            case "baklava":
+                return "https:/\/explorer.celo.org/baklava";
```
Comment (jmrossy) [inline src/lib/CeloscanProvider.ts:30]: These link to Celo's blockscout instance. If the class here is extending `EtherscanProvider`, it should link to [CeloScan](https://celoscan.io/)

Also the escaping of `/` characters is unnecessary

File: test/getHistory.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { CeloscanProvider } from "../src/lib/CeloscanProvider";
+
+async function main() {
+    const account = process.env.ACCOUNT;
+    if (!account) throw new Error("No ACCOUNT provided in env");
+    const network = process.env.NETWORK?.toLocaleLowerCase() || "celo";
+    if (network !== "celo" && network !== "alfajores" && network !== "baklava") throw new Error("Invalid NETWORK provided in env. Use celo, alfajores, or baklava");
```
Comment (jmrossy) [inline test/getHistory.ts:7]: You can let the check in `getBaseUrl` handle this right?

**Review by jmrossy — COMMENTED** [review]
Nice work! Just a few small things to fix and then I'll merge this  and publish an update :)

File: src/lib/CeloscanProvider.ts (hunk 1 above)
Comment (anarkrypto) [reply src/lib/CeloscanProvider.ts:30]: @jmrossy Agree. 

For mainnet: https://api.celoscan.io
Alfajores: https://alfajores.celoscan.io

But https://baklava.celoscan.io is not available for baklava. Any idea ?

File: test/getHistory.ts (hunk 2 above)
Comment (anarkrypto) [reply test/getHistory.ts]: done

File: src/lib/CeloscanProvider.ts (hunk 1 above)
Comment (jmrossy) [reply src/lib/CeloscanProvider.ts:30]: I don't think there's a celoscan instance for baklava so I'd recommend throwing an error for that network. If you'd prefer to keep baklava support in this provider, I'm okay with using blockscout for that network

**Review by jmrossy — APPROVED** [review]

**Pull request merged**